From fd7575cbd24f9f72447755995cd9e6694e23f149 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Fri, 19 Oct 2018 08:02:12 +0200
Subject: [PATCH] anaconda: don't force non-encrypted /boot on coreboot systems
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With grub payload it is possible to have all the partitions encrypted.

Based on patch by @tlaurion

Fixes QubesOS/qubes-issues#2118

Signed-off-by: Frédéric Pierret <frederic.epitre@orange.fr>
---
 pyanaconda/bootloader.py | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/pyanaconda/bootloader.py b/pyanaconda/bootloader.py
index 963af46f9..fa56b532a 100644
--- a/pyanaconda/bootloader.py
+++ b/pyanaconda/bootloader.py
@@ -43,6 +43,8 @@ from blivet.size import Size
 from pyanaconda.i18n import _, N_
 
 import logging
+import subprocess
+
 log = logging.getLogger("anaconda")
 
 class serial_opts(object):
@@ -1411,16 +1413,15 @@ class GRUB2(GRUB):
                           raid.RAID5, raid.RAID6, raid.RAID10]
     stage2_raid_metadata = ["0", "0.90", "1.0", "1.2"]
 
-    @property
-    def stage2_format_types(self):
-        if productName.startswith("Red Hat "):              # pylint: disable=no-member
-            return ["xfs", "ext4", "ext3", "ext2", "btrfs"]
-        else:
-            return ["ext4", "ext3", "ext2", "btrfs", "xfs"]
+    stage2_format_types = ["ext4", "ext3", "ext2", "btrfs", "xfs"]
 
     def __init__(self):
         super(GRUB2, self).__init__()
 
+        self.encryption_support = True
+        self.stage2_format_types += ["lvmlv"]
+        self.skip_bootloader = flags.cmdline.getbool("skip_grub", False)
+
     # XXX we probably need special handling for raid stage1 w/ gpt disklabel
     #     since it's unlikely there'll be a bios boot partition on each disk
 
-- 
2.14.4

